### ISSUE TITLE
Add code_topic() function to code_agreements()

### ISSUE THREAD

**Issue by henriquesposito** [globalgov/qCreate#3]
Add a more detailed code for topics with further explanations about the topics and where their codes come from an why. This should be done in an external file. Currently we are not using topic on code_agreements()

The code_area() function needs also to be re-added as we develop code_lineage() function and plots.

**Comment by henriquesposito** [comment]
I think that the code_topic() function in gnevar might be too specific for what I would personally think of a topic in IR or of an agreement. For example, I would think of IR topics as war/conflict, environment, trade, health or human rights. While what we currently have in gnevar resemble more a combination of issue-domain to me.  

Both approaches might be helpful depending on the level we are interested in looking at, and as long as we define and operationalize topic or issue-domain consistently. 

What do you think @EstherPeev? Should we focus on issue-domain or topic at first and how can we approach this in a more systematic fashion?

**Comment by EstherPeev** [comment]
Yes, I think it could be more useful to have qIDs differentiating the treaties precisely instead of broad topics like you mentioned, at least for now. It will allow to add meaningful information as the topic otherwise would be environment for all of them. 

It means that we could create another function like `code_issue()`, and develop the list of issues we find in environmental treaties. We can also keep `code_topic()`for future differentiation when other qPackages are developed like qHealth, qTrade, qHumanRights, etc

**Comment by henriquesposito** [comment]
We need to define issue then, what is an issue? 

What I am thinking is a combination of issue and subject for topic maybe. Whereas issue implies a subdivision of a subject. The two could be included in a three letter abbreviation (one letter for subject and two for issue).  

For subject we include broader IR concern areas as security, environment, trade, human rights and health. Within these we may want to further subdivide them. For example for the environment, we can subdivide in deforestation, water, climate change and pollution. 

Though we run into problems with agreements that follow into one or more subjects and one or more issues from their title. Either these categories would need to be mutually exclusive or we need to make clear that each follows in to one or more subjects/issues.

**Comment by jhollway** [comment]
Putting a pin in this as current approach is different
